packages/amplify-graphql-relational-transformer/src/utils.ts

```diff
@@ -117,6 +144,26 @@ export function ensureFieldsArray(config: HasManyDirectiveConfiguration | HasOne
   } else if (config.fields.length === 0) {
     throw new InvalidDirectiveError(`No fields passed to @${config.directiveName} directive.`);
   }
+
+  if (config.references) {
+    throw new InvalidDirectiveError(`DynamoDB models do not support 'references' on @${config.directiveName} directive.`);
```

Comment:
This can be checked first so we can fail fast without running the above checks on `fields`.